plasmapy/analysis/swept_langmuir/ion_saturation_current.py

```diff
@@ -0,0 +1,240 @@
+"""
+Functionality for determining the ion-saturation current of a Langmuir sweep.
+"""
+__all__ = ["find_ion_saturation_current", "ISatExtras"]
+__aliases__ = ["find_isat_"]
+
+import numbers
+import numpy as np
+
+from typing import Any, Dict, NamedTuple, Optional, Tuple
+
+from plasmapy.analysis import fit_functions as ffuncs
+from plasmapy.analysis.swept_langmuir.helpers import check_sweep
+
+__all__ += __aliases__
+
+
+class ISatExtras(NamedTuple):
+    """
+    Create a `tuple` containing the extra parameters calculated by
+    `find_ion_saturation_current`.
+    """
+
+    rsq: Optional[float]
+    """
+    Alias for field number 0, the r-squared value of the ion-saturation
+    curve fit.
+    """
+
+    fitted_func: Optional[ffuncs.AbstractFitFunction]
+    """
+    Alias for field number 1, the :term:`fit-function` fitted during
+    the ion-saturation curve fit.
+    """
+
+    fitted_indices: Optional[slice]
+    """
+    Alias for field number 2, the indices used in the ion-saturation
+    curve fit.
+    """
+
+
+def find_ion_saturation_current(
+    voltage: np.ndarray,
+    current: np.ndarray,
+    *,
+    fit_type: str = "exp_plus_linear",
+    current_bound: numbers.Real = None,
+    voltage_bound: numbers.Real = None,
+) -> Tuple[ffuncs.Linear, ISatExtras]:
+    """
+    Determines the ion-saturation current (:math:`I_{sat}`) for a given
+    current-voltage (IV) curve obtained from a swept Langmuir probe.
+    The current collected by a Langmuir probe reaches ion-saturation
+    when the probe is sufficiently biased so the influx of electrons is
+    completely repelled, which leads to only the collection of ions.
+    (For additional details see the **Notes** section below.)
+
+    **Aliases:** `find_isat_`
+
+    Parameters
+    ----------
+
+    voltage: `numpy.ndarray`
+        1-D numpy array of monotonically increasing probe biases
+        (should be in volts).
+
+    current: `numpy.ndarray`
+        1-D numpy array of probe current (should be in amperes)
```

Comment:
```suggestion
        (must be in volts).

    current: `numpy.ndarray`
        1-D numpy array of probe current (must be in amperes)
```
Same optional comment as above.